src/robotFactory.js

```diff
@@ -1,16 +1,25 @@
 'use strict';
 
 class BaseRobot {
-  constructor(name, weight, coords = {}, chipVersion) {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0, y: 0,
+    },
+    chipVersion
+  ) {
     this.name = name;
     this.weight = weight;
 
-    const { x = 0, y = 0 } = coords;
+    if (coords.x === undefined) {
+      coords.x = 0;
+    }
```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```
I'm not sure if checking for undefined for coords x and y is necessary here as though we assign default values in property assignment.